core/src/apps/management/apply_settings.py

```diff
@@ -10,11 +10,32 @@
 if TYPE_CHECKING:
     from trezor.enums import SafetyCheckLevel
     from trezor.messages import ApplySettings, Success
+    from trezor.ui import ProgressLayout
 
 
 BRT_PROTECT_CALL = ButtonRequestType.ProtectCall  # CACHE
 
 
+async def _load_homescreen(length: int) -> bytes:
+    from trezor import utils, workflow
+    from trezor.ui.layouts.progress import progress
+
+    from apps.common import chunked
+
+    loader: ProgressLayout | None = None
+
+    def report(value: int) -> None:
+        nonlocal loader
+        if loader is None:
+            workflow.close_others()
+            loader = progress(TR.progress__please_wait)
```

Comment:
Why not to create the progress layout before calling `get_all_chunks`, instead of launching it on-demand here?

Reply:
It's to handle `length == 0`, but I rewrote it to handle this case explicitly in fd1419a708062560e063c3d26ac3e20da93520ac